File: Exp7.r
# KHAN MOHD OWAIS RAZA
# 20BCD7138
library(ggplot2)
library(dplyr)
data("iris")
iris_data <- iris[, -5] 
scaled_iris <- scale(iris_data)
set.seed(123)
k <- 3 
kmeans_model <- kmeans(scaled_iris, centers = k, nstart = 25)
iris_data$cluster <- as.factor(kmeans_model$cluster)
ggplot(iris_data, aes(x = Petal.Length, y = Petal.Width, color = cluster)) +
  geom_point(size = 3) +
  labs(title = "Clustering of Iris Dataset (Petal Length vs. Petal Width)",
       x = "Petal Length", y = "Petal Width") +
  theme_minimal() +
  scale_color_brewer(palette = "Set1")

File: Exp7.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.datasets import load_iris
from sklearn.cluster import KMeans
from sklearn.preprocessing import StandardScaler
iris = load_iris()
iris_data = pd.DataFrame(data=iris.data, columns=iris.feature_names)
scaler = StandardScaler()
scaled_iris = scaler.fit_transform(iris_data)
k = 3  
kmeans_model = KMeans(n_clusters=k, random_state=123)
kmeans_model.fit(scaled_iris)
iris_data['cluster'] = kmeans_model.labels_
plt.figure(figsize=(8, 6))
plt.scatter(iris_data['petal length (cm)'], iris_data['petal width (cm)'],
            c=iris_data['cluster'], cmap='viridis', edgecolor='k', s=50)
plt.title('Clustering of Iris Dataset (Petal Length vs. Petal Width)')
plt.xlabel('Petal Length (cm)')
plt.ylabel('Petal Width (cm)')
plt.show()
